src/cli/cli_dataset.cpp

```diff
@@ -355,5 +355,208 @@ ThreadError Dataset::ProcessPendingTimestamp(int argc, char *argv[])
+            }
+
+            if (strcmp(argv[index], "networkname") == 0)
+            {
+                VerifyOrExit(index < argc, error = kThreadError_Parse);
+                dataset.mIsNetworkNameSet = true;
+                VerifyOrExit((length = static_cast<int>(strlen(argv[++index]))) <= OT_NETWORK_NAME_MAX_SIZE,
+                             error = kThreadError_Parse);
+                memset(&dataset.mNetworkName, 0, sizeof(sDataset.mNetworkName));
+                memcpy(dataset.mNetworkName.m8, argv[0], static_cast<size_t>(length));
+            }
+
+            if (strcmp(argv[index], "extpanid") == 0)
+            {
+                VerifyOrExit(index < argc, error = kThreadError_Parse);
+                dataset.mIsExtendedPanIdSet = true;
+                VerifyOrExit(Interpreter::Hex2Bin(argv[++index], dataset.mExtendedPanId.m8,
+                                                  sizeof(dataset.mExtendedPanId.m8)) >= 0, error = kThreadError_Parse);
+            }
+
+            if (strcmp(argv[index], "localprefix") == 0)
+            {
+                VerifyOrExit(index < argc, error = kThreadError_Parse);
+                dataset.mIsMeshLocalPrefixSet = true;
+                SuccessOrExit(error = otIp6AddressFromString(argv[++index], &prefix));
+                memcpy(dataset.mMeshLocalPrefix.m8, prefix.mFields.m8, sizeof(dataset.mMeshLocalPrefix.m8));
+            }
+
+            if (strcmp(argv[index], "delaytimer") == 0)
+            {
+                VerifyOrExit(index < argc, error = kThreadError_Parse);
+                dataset.mIsDelaySet = true;
+                SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
+                dataset.mDelay = static_cast<uint32_t>(value);
+            }
+
+            if (strcmp(argv[index], "panid") == 0)
+            {
+                VerifyOrExit(index < argc, error = kThreadError_Parse);
+                dataset.mIsPanIdSet = true;
+                SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
+                dataset.mPanId = static_cast<otPanId>(value);
+            }
+
+            if (strcmp(argv[index], "channel") == 0)
+            {
+                VerifyOrExit(index < argc, error = kThreadError_Parse);
+                dataset.mIsChannelSet = true;
+                SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
+                dataset.mChannel = static_cast<uint16_t>(value);
+            }
+
+            if (strcmp(argv[index], "binary") == 0)
+            {
+                VerifyOrExit((index + 1) < argc, error = kThreadError_Parse);
+                SuccessOrExit(error = Interpreter::ParseLong(argv[++index], value));
+                VerifyOrExit(Interpreter::Hex2Bin(argv[++index], tlvs + size, static_cast<uint16_t>(value)) >= 0,
+                             error = kThreadError_Parse);
+                size += value;
+            }
+
```

Comment:
Check for parsing errors?  For example, a argument that does not match any of the cases above?